### PR TITLE
Optimize circleci file

### PR DESCRIPTION
**Pull request by dragonsahead**
This PR will:
- reduce the number of lines in the CircleCI config by using matrixes
- use smaller resource types to for cost cutting (we don't need medium machines for some tasks, small ones are more than enough)
- use one task for driver building, to reduce time

### CODE REVIEW

**Review by nemanjaglumac — CHANGES_REQUESTED** [review]
Please rebase on the latest`master`. There was at least one new e2e CI group added ("filters"), and we are now also running smoke tests as a "regular" e2e CI group.

File: .circleci/config.yml [hunk 1]
```diff
@@ -1173,111 +1140,12 @@ workflows:
...
-            - build-uberjar-<< matrix.edition >>
-          cypress-group: "sharing-<< matrix.edition >>"
-          source-folder: frontend/test/metabase/scenarios/sharing
-
-      - fe-tests-cypress:
-          matrix:
-            parameters:
-              edition: ["ee", "oss"]
-          name: e2e-tests-visualizations-<< matrix.edition >>
+              test: ["admin", "binning", "collections", "dashboard", "onboarding", "native", "question", "sharing", "visualizations"]
```
Comment (nemanjaglumac) [inline .circleci/config.yml:1143]:
```suggestion
              test: ["admin", "binning", "collections", "dashboard", "filters", "onboarding", "native", "question", "sharing", "smoketests", "visualizations"]
```

**Review by nemanjaglumac — APPROVED** [review]
🎉

**Comment by paoliniluis** [comment]
can't merge as it seems that there are some status reports that aren't completed due to the name changes, so if @camsaul can merge it would be great! (BTW: all steps green)

**Comment by paoliniluis** [comment]
I'm going to close this as this would be implemented in @nemanjaglumac 's PR

**Comment by nemanjaglumac** [comment]
Oh no, you shouldn't.
My PR deals oy with the Cypress changes.

We could still benefit from other matrices.

Please reopen and do your magic 🪄

**Comment by paoliniluis** [comment]
oh! thanks, thought that it was imported in your PR, thanks!

**Review by camsaul — APPROVED** [review]
WFM 👍

**Comment by paoliniluis** [comment]
This one too is ready to rumble, once we fix the "/api/login-history/" endpoint issues

**Comment by paoliniluis** [comment]
@camsaul is the strategy of building just 1 time the drivers and reusing those for both uberjars OK?

**Review by nemanjaglumac — CHANGES_REQUESTED** [review]
This PR has been open for far too long.
On top of that, it's touching a lot of different areas and should be broken down into several individual smaller PRs for easier damage control.
- executors
- cache
- resource allocation
- updated requirements and version updates
- matrix refactor (simplification of long and repetitive code)

**Comment by paoliniluis** [comment]
I'm closing this PR to separate it between:
- https://github.com/metabase/metabase/pull/19489
- https://github.com/metabase/metabase/pull/19490

using smaller resources needs some more investigation now that CircleCI supports seeing resources in the process

**Pull request closed without merging**